SSA/Core/MLIRSyntax/GenericParser.lean

```diff
@@ -549,6 +549,13 @@ syntax "@" str : mlir_attr_val_symbol
 syntax "#" ident : mlir_attr_val -- alias
 syntax "#" strLit : mlir_attr_val -- aliass
 
+declare_syntax_cat dialect_attribute_contents
+syntax mlir_attr_val : dialect_attribute_contents
+syntax "(" dialect_attribute_contents + ")" : dialect_attribute_contents
+syntax "[" dialect_attribute_contents + "]": dialect_attribute_contents
+syntax "{" dialect_attribute_contents + "}": dialect_attribute_contents
+syntax "#" ident "<" dialect_attribute_contents ">" : mlir_attr_val -- opaqueAttr
+
```

Comment:
What is a `dialect_attribute_contents`, and how does it differ from a regular `mlir_attr_val`?

Since there seems to be quite some discussion around this PR already, I'd suggest taking the changes to the parser, which seem somewhat independent, out of this minimal PR.

Reply:
Please don't hit unresolve unless its uncontroversial. I'd still say these *syntactic* changes (how are overflow flags represented in the DSL) are done separately from the main *semantic* change of (optionally) having UB on overflow.

That said, I won't die on this hill. If you're convinced this is the right approach, feel free to go ahead. I would still like to see a docstring explain that `dialect_attribute_contents` is a dialect-namespaced `mlir_attr_val`